tests/integration/test_exodus.py

```diff
@@ -0,0 +1,81 @@
+import requests
+
+
+def test_exodus_basic(cdn_test_url):
+    url = (
+        cdn_test_url
+        + "/content/beta/rhel8/8/x86_64/exodus_test/os/bash-4.4.19-7.el8.x86_64.rpm"
+    )
+
+    r = requests.get(url)
+    assert r.status_code == 200
+    assert "cache-control" not in r.headers
+
+
+def test_header_not_exist_file(cdn_test_url):
+    url = (
+        cdn_test_url
+        + "/content/beta/rhel8/8/x86_64/exodus_test/os/bash-4.4.19-7.el8.x86_64.rpm_no"
+    )
+    r = requests.get(url)
+    assert r.status_code == 404
+    assert "cache-control" not in r.headers
+
+
+def test_header_cache_control(cdn_test_url):
+    patterns = [
+        "/content/beta/rhel8/8/x86_64/exodus_test/os/listing",
+        "/content/beta/rhel8/8/x86_64/exodus_test_ctype/os/repodata/repomd.xml",
+        "/content/beta/rhel8/8/x86_64/exodus_test/os/ostree/repo/refs/heads/sub_dir/testing",
+    ]
+    multiple_url = [cdn_test_url + pattern_item for pattern_item in patterns]
+    for url in multiple_url:
+        r = requests.get(url)
+        assert r.status_code == 200
+        assert r.headers["cache-control"] == "max-age=600"
+
+
+def test_header_want_digest_get(cdn_test_url):
+    headers = {"want-digest": "sha256"}
```

Comment:
This isn't the want-digest we expect users to pass, they should pass id-sha-256.  It doesn't actually make a difference to the test because we still return id-sha-256 anyway.  I'd prefer if you could provide the real checksum type though.

Actually I'm getting a bit nervous about our behavior of returning id-sha-256 even if the user passed something wrong; I might file an issue requesting to change that.